app/src/main/java/me/saket/dank/ui/accountmanager/AccountManagerActivity.java

```diff
@@ -0,0 +1,290 @@
+import me.saket.dank.data.ResolvedError;
+import me.saket.dank.di.Dank;
+import me.saket.dank.ui.DankActivity;
+import me.saket.dank.ui.authentication.LoginActivity;
+import me.saket.dank.ui.user.UserSessionRepository;
+import me.saket.dank.ui.accountmanager.AccountManagerAdapter.AccountManagerViewHolder;
+import me.saket.dank.ui.subscriptions.SubscriptionRepository;
+import me.saket.dank.utils.ItemTouchHelperDragAndDropCallback;
+import me.saket.dank.utils.RxDiffUtil;
+import me.saket.dank.utils.itemanimators.SlideUpAlphaAnimator;
+import me.saket.dank.widgets.swipe.RecyclerSwipeListener;
+
+@DeepLink(AccountManagerActivity.DEEP_LINK)
+public class AccountManagerActivity extends DankActivity {
+  public static final String DEEP_LINK = "dawn://accountManager";
+
+  private final int ACTION_DELETE = 1;
+  private final int ACTION_LOGOUT = 2;
+
+  private AccountManager selectedAccount = null;
+
+  @BindView(R.id.account_manager_root) ViewGroup rootViewGroup;
+  @BindView(R.id.account_manager_accounts_recyclerview) RecyclerView accountsRecyclerView;
+  @BindView(R.id.account_manager_logout) Button logoutButton;
+  @BindView(R.id.account_progressbar) View loadingProgessbar;
+
+  @Inject Lazy<UserSessionRepository> userSessionRepository;
+  @Inject Lazy<SubscriptionRepository> subscriptionRepository;
+  @Inject Lazy<AccountManagerRepository> accountManagerRepository;
+  @Inject Lazy<AccountManagerAdapter> accountManagerAdapter;
+  @Inject Lazy<ErrorResolver> errorResolver;
+
+  private Disposable confirmTimer = Disposables.disposed();
+  private Disposable timerDisposable = Disposables.empty();
+
+  @CheckResult
+  public static Intent intent(Context context) {
+    return new Intent(context, AccountManagerActivity.class);
+  }
+
+  @Override
+  protected void onCreate(@Nullable Bundle savedInstanceState) {
+    Dank.dependencyInjector().inject(this);
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_account_manager);
+    ButterKnife.bind(this);
+    setupUserList();
+  }
+
+  @Override
+  @OnClick(R.id.account_manager_done)
+  public void finish() {
+    timerDisposable.dispose();
+    super.finish();
+  }
+
+  @OnClick(R.id.account_manager_logout)
+  public void logout() {
+    confirmAction(ACTION_LOGOUT);
+  }
+
+  private Completable queueToDelete(AccountManager account){
+    this.selectedAccount = account;
+    return this.confirmAction(ACTION_DELETE);
+  }
+
+  private Completable confirmAction(int action) {
+    if (confirmTimer.isDisposed()) {
+      int confirmText = ACTION_LOGOUT == action ? R.string.userprofile_confirm_logout : R.string.userprofile_confirm_delete;
+
+      runOnUiThread(() -> {
+        // Stuff that updates the UI
+        logoutButton.setText(confirmText);
+        logoutButton.setVisibility(View.VISIBLE);
+      });
+
+
+      confirmTimer = Observable.timer(5, TimeUnit.SECONDS)
+          .compose(applySchedulers())
+          .subscribe(o -> {
+            runOnUiThread(() -> {
```

Comment:
Perhaps there is an operator on RX that would make subscribe execute on main thread making this switch redundant?

Reply:
We can use `.subscribeOn(AndroidSchedulers.mainThread())` which this code doesn't seem to do because it already used [RxUtils#applySchedulers](https://github.com/Tunous/Dawn/blob/5bb5366e70bf4c6364f6e50d3f3b8981d740cb3e/app/src/main/java/me/saket/dank/utils/RxUtils.java#L30-L34) to subscribe on the io scheduler.